registry/remote/example_test.go

```diff
@@ -181,6 +185,56 @@ func ExampleRepository_Push() {
 	// Push finished
 }
 
+// ExampleRepository_Push_referenceManifest gives an example snippet for pushing a reference manifest.
+func ExampleRepository_Push_referenceManifest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	// 1. assemble the referenced manifest
+	rfedManifest := artifactspec.Manifest{
+		MediaType:    artifactspec.MediaTypeArtifactManifest,
+		ArtifactType: "referenced manifest",
+	}
+	rfedManifestContent, _ := json.Marshal(rfedManifest)
+	rfedManifestDescriptor := artifactspec.Descriptor{
+		MediaType:    artifactspec.MediaTypeArtifactManifest,
+		ArtifactType: "referenced manifest descriptor",
+		Digest:       ocidigest.FromBytes(rfedManifestContent),
+		Size:         int64(len(rfedManifestContent)),
+	}
+
+	// 2. assemble the referencing artifact manifest
+	rfingManifest := artifactspec.Manifest{
+		MediaType:    artifactspec.MediaTypeArtifactManifest,
+		ArtifactType: "referencing manifest",
+		Subject:      rfedManifestDescriptor,
+	}
+	rfingManifestContent, _ := json.Marshal(rfingManifest)
+	rfingManifestDescriptor := artifactspec.Descriptor{
+		MediaType:    artifactspec.MediaTypeArtifactManifest,
+		ArtifactType: "referencing manifest descriptor",
+		Digest:       ocidigest.FromBytes(rfingManifestContent),
+		Size:         int64(len(rfingManifestContent)),
+	}
+
+	// 3. push the manifest descriptor and content
+	err = repo.Push(ctx, descriptor.ArtifactToOCI(rfingManifestDescriptor), bytes.NewReader(rfingManifestContent))
```

Comment:
`oras.land/oras-go/v2/internal/descriptor` is an internal package. It should never be used in the examples.

Reply:
removed this function and converted the types manually.